File: TripKitAndroid/src/main/java/com/skedgo/tripkit/tsp/TspModule.java
package com.skedgo.tripkit.tsp;

import com.google.gson.Gson;

import dagger.Module;
import dagger.Provides;
import io.reactivex.schedulers.Schedulers;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;
import skedgo.tripkit.configuration.Server;

@Module
public class TspModule {
  @Provides RegionInfoApi regionInfoApi(
      Gson gson,
      okhttp3.OkHttpClient httpClient) {
    return new Retrofit.Builder()
        .baseUrl(Server.ApiTripGo.getValue())
        .client(httpClient)
        .addConverterFactory(GsonConverterFactory.create(gson))
        .addCallAdapterFactory(RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io()))
        .build()
        .create(RegionInfoApi.class);
  }
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/tsp/TspModule.kt
package com.skedgo.tripkit.tsp

import com.google.gson.Gson
import com.skedgo.tripkit.configuration.ServerManager.configuration
import dagger.Module
import dagger.Provides
import io.reactivex.schedulers.Schedulers
import okhttp3.OkHttpClient
import retrofit2.Retrofit.Builder
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

@Module
class TspModule {
    @Provides
    fun regionInfoApi(
        gson: Gson,
        httpClient: OkHttpClient
    ): RegionInfoApi {
        return Builder()
            .baseUrl(configuration.apiTripGoUrl)
            .client(httpClient)
            .addConverterFactory(GsonConverterFactory.create(gson))
            .addCallAdapterFactory(RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io()))
            .build()
            .create(RegionInfoApi::class.java)
    }
}